src/main/java/uk/gov/companieshouse/company/links/service/AddStatementsClient.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.companieshouse.company.links.service;
+
+import java.util.function.Supplier;
+
+import org.springframework.stereotype.Component;
+
+import uk.gov.companieshouse.api.InternalApiClient;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.company.links.exception.NonRetryableErrorException;
+import uk.gov.companieshouse.company.links.exception.RetryableErrorException;
+import uk.gov.companieshouse.logging.Logger;
+
+@Component
+public class AddStatementsClient implements LinkClient {
+    private final Logger logger;
+    private final Supplier<InternalApiClient> internalApiClientFactory;
+
+    public AddStatementsClient(Logger logger, Supplier<InternalApiClient> internalApiClientFactory) {
+        this.logger = logger;
+        this.internalApiClientFactory = internalApiClientFactory;
+    }
+
+    /**
+     * Sends a patch request to the add statements link endpoint in
+     * the company profile api and handles any error responses.
+     */
+    @Override
+    public void patchLink(String companyNumber) {
+        InternalApiClient client = internalApiClientFactory.get();
+        try {
+            client.privateCompanyLinksResourceHandler()
+                    .addPscStatementsCompanyLink(String.format("/company/%s/links/persons-with-significant-control-statements", companyNumber))
```

Comment:
Could we first get the number of psc statements from mongo and confirm that there is at least one before adding a link?

Use the getStatements method from the sdk to make an api call